src/run-child-proc.cc

```diff
@@ -291,6 +311,56 @@ void base_process_service::run_child_proc(run_proc_params params) noexcept
         if (setrlimit(limit.resource_id, &setlimits) != 0) goto failure_out;
     }
 
+    // nice
+    if (nice_is_set) {
+        err.stage = exec_stage::SET_PRIO;
+        #ifdef __linux__
+        // clamp the values to known range so the autogroup hack below works
+        if (nice > 19) nice = 19;
+        if (nice < -20) nice = -20;
+        #endif
+        if (setpriority(PRIO_PROCESS, getpid(), nice) != 0) goto failure_out;
+        #ifdef __linux__
+        // we usually create a new session leader; that makes nice not very
+        // useful as the Linux kernel will autogroup processes by session id
+        // except when disabled - so also work around this where enabled
+        // the r+ is used in order to avoid creating it where already disabled
+        errno = 0;
+        FILE *ag = std::fopen("/proc/self/autogroup", "r+");
+        if (ag) {
+            std::fprintf(ag, "%d\n", nice);
+            std::fclose(ag);
+        }
+        else if (errno != ENOENT) goto failure_out;
+        #endif
+    }
+
+    #if SUPPORT_IOPRIO
+    // ioprio
+    if (ioprio >= 0) {
+        err.stage = exec_stage::SET_PRIO;
+        if (syscall(__NR_ioprio_set, 1, (int)getpid(), ioprio) != 0) goto failure_out;
+    }
+    #endif
+
+    #if SUPPORT_OOM_ADJ
+    // oom score adjustment
+    if (oom_adj_is_set) {
+        err.stage = exec_stage::SET_PRIO;
+        errno = 0;
+        int fd = open("/proc/self/oom_score_adj", O_WRONLY);
+        if (fd < 0) goto failure_out;
+        // +4: round up, minus sign, newline, nul terminator
+        char val_str[std::numeric_limits<int>::digits10 + 4];
+        int num_chars = snprintf(val_str, sizeof(val_str), "%hd\n", oom_adj);
```

Comment:
`%hd` should be just `%d`, I believe (the value is a plain `int`, not a `short int`).

Reply:
Ah I see. In that case line 354 should have `std::numeric_limits<short>` instead of `std::numeric_limits<int>`.

Reply:
done